airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/job_error_reporter/SentryJobErrorReportingClient.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.scheduler.persistence.job_error_reporter;
+
+import io.airbyte.config.FailureReason;
+import io.airbyte.config.Metadata;
+import io.airbyte.config.StandardWorkspace;
+import io.sentry.Hub;
+import io.sentry.IHub;
+import io.sentry.NoOpHub;
+import io.sentry.SentryEvent;
+import io.sentry.SentryOptions;
+import io.sentry.protocol.Message;
+import io.sentry.protocol.SentryException;
+import io.sentry.protocol.User;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class SentryJobErrorReportingClient implements JobErrorReportingClient {
+
+  static final String STACKTRACE_PARSE_ERROR_TAG_KEY = "stacktrace_parse_error";
+  private final IHub sentryHub;
+  private final SentryExceptionHelper exceptionHelper;
+
+  SentryJobErrorReportingClient(final IHub sentryHub, final SentryExceptionHelper exceptionHelper) {
+    this.sentryHub = sentryHub;
+    this.exceptionHelper = exceptionHelper;
+  }
+
+  public SentryJobErrorReportingClient(final String sentryDSN, final SentryExceptionHelper exceptionHelper) {
+    this(createSentryHubWithDSN(sentryDSN), exceptionHelper);
+  }
+
+  static IHub createSentryHubWithDSN(final String sentryDSN) {
+    if (sentryDSN == null || sentryDSN.isEmpty()) {
+      return NoOpHub.getInstance();
+    }
+
+    final SentryOptions options = new SentryOptions();
+    options.setDsn(sentryDSN);
+    options.setAttachStacktrace(false);
+    options.setEnableUncaughtExceptionHandler(false);
+    return new Hub(options);
+  }
+
+  /**
+   * Reports a Connector Job FailureReason to Sentry
+   *
+   * @param workspace - Workspace where this failure occurred
+   * @param failureReason - FailureReason to report
+   * @param dockerImage - Tagged docker image that represents the release where this failure occurred
+   * @param metadata - Extra metadata to set as tags on the event
+   */
+  @Override
+  public void reportJobFailureReason(final StandardWorkspace workspace,
+                                     final FailureReason failureReason,
+                                     final String dockerImage,
+                                     final Map<String, String> metadata) {
+    final SentryEvent event = new SentryEvent();
+
+    // Remove invalid characters from the release name, use @ so sentry knows how to grab the tag
+    // e.g. airbyte/source-xyz:1.2.0 -> source-xyz@1.2.0
+    // More info at https://docs.sentry.io/product/cli/releases/#creating-releases
+    final String release = dockerImage.replace(":", "@").substring(dockerImage.lastIndexOf("/") + 1);
```

Comment:
+1 to that

Reply:
Done in [0bafd6f](https://github.com/airbytehq/airbyte/pull/13899/commits/0bafd6f6670f1d4e3fedec807522d5f7d2e7010c). 

Because Sentry auto-creates a release if an event comes in with a release that doesn't yet exist and we've been creating releases in the old format when publishing connectors, we'll have "duplicate" releases in Sentry (e.g. airbyte-source-stripe@1.2.3 and source-stripe@1.2.3) if we start sending events from cloud before we update the publish workflow. To make it cleaner, we should probably just use a new sentry project @Phlair

Reply:
Agreed on new project 👍